src/UsgsAstroPushFrameSensorModel.cpp

```diff
@@ -0,0 +1,2461 @@
+                 time, 4, nOrderQuat, q);
+}
+
+//***************************************************************************
+// UsgsAstroLineScannerSensorModel::calculateAttitudeCorrection
+//***************************************************************************
+void UsgsAstroPushFrameSensorModel::calculateAttitudeCorrection(
+    const double& time, const std::vector<double>& adj,
+    double attCorr[9]) const {
+  MESSAGE_LOG(
+      "Computing calculateAttitudeCorrection (with adjustment)"
+      "for time {}",
+      time)
+  double aTime = time - m_t0Quat;
+  double euler[3];
+  double nTime = aTime / m_halfTime;
+  double nTime2 = nTime * nTime;
+  euler[0] = (getValue(6, adj) + getValue(9, adj) * nTime +
+              getValue(12, adj) * nTime2) /
+             m_flyingHeight;
+  euler[1] = (getValue(7, adj) + getValue(10, adj) * nTime +
+              getValue(13, adj) * nTime2) /
+             m_flyingHeight;
+  euler[2] = (getValue(8, adj) + getValue(11, adj) * nTime +
+              getValue(14, adj) * nTime2) /
+             m_halfSwath;
+  MESSAGE_LOG("calculateAttitudeCorrection: euler {} {} {}", euler[0], euler[1],
+              euler[2])
+
+  calculateRotationMatrixFromEuler(euler, attCorr);
+}
+
+//***************************************************************************
+// UsgsAstroPushFrameSensorModel::losToEcf
+//***************************************************************************
+void UsgsAstroPushFrameSensorModel::losToEcf(
+    const double& line,              // CSM image convention
+    const double& sample,            // UL pixel center == (0.5, 0.5)
+    const std::vector<double>& adj,  // Parameter Adjustments for partials
+    double& xc,                      // output sensor x coordinate
+    double& yc,                      // output sensor y coordinate
+    double& zc,                      // output sensor z coordinate
+    double& vx,                      // output sensor x velocity
+    double& vy,                      // output sensor y velocity
+    double& vz,                      // output sensor z velocity
+    double& bodyLookX,               // output line-of-sight x coordinate
+    double& bodyLookY,               // output line-of-sight y coordinate
+    double& bodyLookZ) const         // output line-of-sight z coordinate
+{
+  //# private_func_description
+  // Computes image ray (look vector) in ecf coordinate system.
+  // Compute adjusted sensor position and velocity
+  MESSAGE_LOG(
+      "Computing losToEcf (with adjustments) for"
+      "line {} sample {}",
+      line, sample)
+
+  double time = getImageTime(csm::ImageCoord(line, sample));
+  getAdjSensorPosVel(time, adj, xc, yc, zc, vx, vy, vz);
+
+  // Compute the line within the framelet
+  int summedFrameletHeight = m_frameletHeight / m_detectorLineSumming;
+  double frameletLine = std::fmod(line, summedFrameletHeight);
+  if (m_frameletsFlipped) {
+    frameletLine = summedFrameletHeight - frameletLine;
+  }
```

Comment:
This is the new part of this function. This covers the differences between push frame and line scan for all transformations from image space to object space.